### ISSUE TITLE
Monitorian says DDC not supported because of some glitch - Add a retry button for it

### ISSUE THREAD

**Issue by VarunAgw** [emoacht/Monitorian#149]
Because of some glitches, it says DDC not available, but when I restart the application, it works fine.

I don't blame Monitorian for having glitches, but a "Retry" button will be great. It should show it along with "DDC not available" error. On click, it should try again, which most likely will work.

**Comment by emoacht** [comment]
You can find similar proposal in closed issues.
I personally hate to add such button because this app should deal with such event automatically and am still struggling the way to do so. If you wish to contribute, please provide [operation.log](https://github.com/emoacht/Monitorian#operations) so that I could look into what is really happening.

**Comment by VarunAgw** [comment]
I have enabled operation logs now. Will upload it soon. I don't think a refresh button is such a bad idea (beats restarting the application).

**Comment by NanoExplorer** [comment]
Hey there! Just wanted to report I'm seeing this issue too. Monitorian says neither of my monitors are supported, but only when Monitorian is started automatically (either using the "start on login" option built-in to Monitorian or with a task scheduler event for log on, even if I use task scheduler to delay 30 seconds) but not if I manually start it by double-clicking the shortcut. For now I'll work around by disabling auto-start and just starting it when I need it, but I'll edit this to add my operation log in a sec in case that helps!

For reference, I have two monitors and one of them really doesn't support DDC. The unsupported monitor is connected to my pc over a displayport/vga bridge, and the supported monitor is connected via HDMI.

**Comment by Lechenx** [comment]
I have the same issue. Sometimes the software needs to reboot several times to detect the monitor correctly.
Here is the [operation.log](https://gist.githubusercontent.com/Lechenx/34e3321a3daccdb336ea7301085bd4fd/raw/33e7cca873d81300c5c540fe331a5723ae5c36ae/operation.log) , which you can see the the first two start up failed to recognize my monitor, only the last try works.

**Comment by emoacht** [comment]
@NanoExplorer, @Lechenx Thank you for the log.
I understand there are monitors that are sluggish at bootup. I have never imagined they can be so surprisingly slow.
The next question is how long do they take to be fully normal. More than 5 or 7 minutes? I have no idea how to treat such slow ones.

**Comment by emoacht** [comment]
@NanoExplorer, @Lechenx Thank you for the log.
I understand there are monitors that are sluggish at bootup. I have never imagined they can be so surprisingly slow.
The next question is how long does it take to become fully normal. More than 5 or 7 minutes?
I have no idea how to treat such slow monitors.

**Comment by Lechenx** [comment]
> @NanoExplorer, @Lechenx Thank you for the log.
> I understand there are monitors that are sluggish at bootup. I have never imagined they can be so surprisingly slow.
> The next question is how long does it take to become fully normal. More than 5 or 7 minutes?
> I have no idea how to treat such slow monitors.

Sorry, I do not quiet understand what you mean. This issue seems does not matter with time. When I finall get it works normal, once if I turn it down and reopen, there possibly still a warning says ddc/ci not supported.

**Comment by emoacht** [comment]
@Lechenx Do you mean that the log is not at the time of bootup? If so, it is not the same issue as @NanoExplorer's and I have no clue why your monitor failed to respond to the first two scans.

**Comment by Lechenx** [comment]
> @Lechenx Do you mean that the log is not at the time of bootup? If so, it is not the same issue as @NanoExplorer's and I have no clue why your monitor failed to respond to the first two scans.

Yes, that's what I mean. It's a pity that I have to choose another program, your work is the most convenient ones in those I have used. 
Thanks for your help!

**Comment by MPeti1** [comment]
I have also had the same problem for a few days now, just enabled operation logging.
I have Windows 10 1909 and an LG 22MB37, if it helps. Maybe if others publish what Windows 10 version and monitor they use, we might find some correlation

**Comment by MPeti1** [comment]
Since my last comment (when I enabled logging) it's the first occasion when I noticed that Monitorian can't find my monitor, so I saved the logs and here is what have I found.

For the first time it occurred after a hibernation. It seems it still knows about the monitor, but it's stored with a different type which is probably not listed on the monitors list.
Log entries from when `UnreachableMonitorItem` type of items were first logged:
https://gist.github.com/MPeti1/fc436c0dbb54fa0f8234ee6184fba1d9#file-first-occurrence-of-error

A day later Monitorian was able to find my monitor for once again, but not anymore: that's because on that day I rebooted my PC.
https://gist.github.com/MPeti1/fc436c0dbb54fa0f8234ee6184fba1d9#file-monitor-was-found-for-once-again-but-not-more

It seems like if Monitorian would receive monitors in a bad state if it has been running for some time, which time can vary because for the first time it worked flawlessly for days, but last time it got broken in a day.
What I'm almost sure about though, is that it always happens after a hibernation. I can't test it right now, but I'll soon try it out.

PS: for privacy reasons I only included pieces of the log which seemed relevant, but if it's not enough I can send you full logs in some kind of direct message.

**Comment by MPeti1** [comment]
Also, rescanning for monitors does not help, but probing into monitors makes my monitor dim brightly for a short while (it's currently with reduced brightness)

**Comment by emoacht** [comment]
@MPeti1 Thank you for the logs.
"UnreachableMonitorItem" indicates that the monitor is connected but does not respond to DDC/CI command. According to the past reports, it is quite common after resume. Usually it will not last long and not noticeable. The point is how long does it take to return back to normal state or never?
I made some tweaks in Ver 2.14 and changed to record error code to operation.log. I am not sure it would produce meaningful information though.

**Comment by MPeti1** [comment]
> he point is how long does it take to return back to normal state or never?

Most of the times it works right after a resume, but when it does not then it seems to be stuck indefinitely, or until I restart Monitorian. There were cases where I left the brightness at the minimum instead of restarting monitorian and adjusting, to see if it comes back by itself, but it didn't. When this happens, restarting Monitorian (and doing nothing else, as far as I'm aware) makes it work immediately.

**Comment by MPeti1** [comment]
Could you also relase the executables in itself? I use Monitorian that way.
That might have been better if I say it earlier, in case it can have it's side effects, but I forgot to mention it.

**Comment by emoacht** [comment]
As part of simplifying release process, I am going to release installer file only.
You can still extract executable files without side effects by following command.

    msiexec /a [source msi file path] targetdir=[destination folder path] /qn

**Comment by MPeti1** [comment]
Thank you!
For anyone reading this in the future, you need to give an absolute path to targetdir.

**Comment by emoacht** [comment]
I am not sure what you suggest on targetdir. It's the folder path where a user wish to extract executable files. The absolute path depends on each user.

**Comment by MPeti1** [comment]
Earlier today I was seeing an open window of the Monitorian popup in the alt+tab window switcher, and it showed an error that in multiple attempts it wasn't able to find a controllable monitor, but it was in the way so I closed it, and it might have either compeletely exited Monitorian, or it crashed, because I haven't seen it's tray icon anymore.

Anyway, I was able to export the logs, here I pasted the last successful monitor detection, the first unsuccessfull one, and everything in between: https://gist.github.com/MPeti1/1a8da787b56161c39cd836324222da96

**Comment by emoacht** [comment]
Thank you for the report.
It is worth to know what types of error occur in failure case. If you find other error code and message, please post it as well.

**Comment by MPeti1** [comment]
It's been a long time since I last responded, so i wanted to let you know that I haven't forgot about this issue, but I didn't have any problems with Monitorian since the last update.

**Comment by therealjohnsnow** [comment]
Hi emoacht, unfortunately i am still struggling with this issue.

have attached probe log and operation log for your kind perusal. 
[probe.log](https://github.com/emoacht/Monitorian/files/6809786/probe.log)
[operation.log](https://github.com/emoacht/Monitorian/files/6809787/operation.log)

For context, this is a 1 monitor, connected via vga to work pc in the day, and (dvi-hdmi) to my personal laptop at night. so far the issue mainly occurs on the home pc side only. work pc monitorian works fairly well.

pls let me know what you find. thank you
